### PR TITLE
Verify Email after signup

### PR DESCRIPTION
**Pull request by Bhawna793**

### CODE REVIEW

File: Backend/controllers/product.js [hunk 1]
```diff
@@ -1,113 +1,128 @@
...
+    if (
+      !req.body.name ||
+      !req.body.price ||
+      !req.body.description ||
+      !req.body.category ||
+      !req.body.subcategory ||
+      !req.body.discount ||
+      !req.body.Quantity 
+    ) {
+        console.log(req.body)
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js]: Remove consoles

File: Backend/controllers/product.js [hunk 2]
```diff
@@ -1,113 +1,128 @@
...
-async function getProduct(req,res){
-    try {
-        const product = await products.find();
-        res.json(product);
-    } catch (error) {
-        res.status(500).json({ error: 'An error occurred while fetching products' });
-        console.log(error)
+    const existingProduct = await products.findOne({ name: req.body.name });
+    if (existingProduct) {
+      console.log("Product already exists");
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js]: same here.

File: Backend/controllers/product.js [hunk 3]
```diff
@@ -1,113 +1,128 @@
...
+      return res.status(400).json({ msg: "Product already exists" });
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js:24]: Any specific reason for using return

File: Backend/controllers/product.js [hunk 4]
```diff
@@ -1,113 +1,128 @@
...
+    }
+
+    const priceValidationRegex = /^[0-9]{1,10}(\.\d{1,3})?$/;
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js]: Move validation at the beginning of the controller.

File: Backend/controllers/product.js [hunk 5]
```diff
@@ -1,113 +1,128 @@
...
+    if (!quantityValidationRegex.test(req.body.Quantity)) {
+      console.log("Invalid quantity");
+      return res.status(400).json({
+        msg: "Invalid Quantity",
+      });
+    }
+
+    const currUser = getUser(req.cookies.accessToken);
+
+    console.log(currUser);
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js]: remove consoles.

File: Backend/controllers/product.js [hunk 6]
```diff
@@ -1,113 +1,128 @@
...
-        const userId = data.curUser._id;
-        const product = await products.find({createdBy: userId});
-        res.json(product);
-    } catch (error) {
-        res.status(500).json({ error: 'An error occurred while fetching products'});
-        console.log(error);
-    }
+  try {
+    const accessToken = req.cookies?.accessToken;
+    const data = getUser(accessToken);
```
Comment (DaffodilAqib) [inline Backend/controllers/product.js]: Do this part in authentication middleware.

File: Backend/controllers/user.js [hunk 7]
```diff
@@ -1,219 +1,291 @@
...
-
 
-async function handleUserLogin(req, res, next) {
-    try {
-        const {email, password} = req.body;
-        const curUser = await user.findOne({email});
-        if(!curUser) {
-            return res.status(400).json({msg: "Invalid Email"});
-        }
+    const existingUser =
```
Comment (DaffodilAqib) [inline Backend/controllers/user.js]: You can write this expression as const existingUser = await user.findOne({
  $or: [{ email }, { mobile }]
});

File: Backend/controllers/user.js [hunk 8]
```diff
@@ -1,219 +1,291 @@
...
+      (await user.findOne({ email })) || (await user.findOne({ mobile }));
+    if (existingUser) {
+      return res.status(400).json({ msg: "User already exists" });
+    }
 
-        const result = await bcrypt.compare(password, curUser.password)
+    const passwordValidationRegex =
```
Comment (DaffodilAqib) [inline Backend/controllers/user.js:20]: Move the validation part to the beginning of the controller.

File: Backend/controllers/product.js (hunk 3 above)
Comment (Bhawna793) [inline Backend/controllers/product.js:57]: for returning to controller we need this

File: Backend/controllers/user.js [hunk 9]
```diff
@@ -1,219 +1,291 @@
...
-  
-    } catch (err) {
-      res.status(500).send({ message: 'Error sending reset email' });
+  const { email } = req.body;
+  try {
+    if (!email) {
+      return res.status(403).json({ msg: "Please fill your email" });
     }
+
+    const user1 = await user.findOne({ email });
```
Comment (DaffodilAqib) [inline Backend/controllers/user.js]: We can use "findOneAndUpdate" method here.

**Pull request closed without merging**